### ISSUE TITLE
--unrestricted, --all-text, etc cause --ignore to not be obeyed

### ISSUE THREAD

**Issue by ggreer** [ggreer/the_silver_searcher#245]
Probably need to change the order in which those are parsed, or stick things in --ignore into hardcoded ignores. Thanks to @locks for reporting this.

**Comment by locks** [comment]
poke, is this still relevant?

**Comment by rosshadden** [comment]
Is there a suggested workaround?

**Comment by seguri** [comment]
I'm currently using this as a workaround:

```
ag -G '[^\.svn]' PATTERN
```

instead of something like:

```
ag -u --ignore=.svn PATTERN
```

I also thought of using specific `.agignore`s files. Like building an alias:

```
# "ag ignore"
agi() {
  # find a way to read ignore pattern
  # put pattern in a temporary .agignore, see mktemp
  ag -p /path/to/.agignore "$@"
  # delete temp file
}
```

Not tested, just an idea.

**Comment by rosshadden** [comment]
@seguri I'm not sure if your workaround is working as you expect.  `[^\.svn]` would match anything that doesn't have a `.`, `s`, `v`, or `n` everywhere, which would mean everything would pass because even files with `.` or `s` in their paths have other characters that are not in that character group.

**Comment by seguri** [comment]
@rosshadden Yup you're right, that was just a (wrong) idea. I just used this one to test if `-u` and `-G` together are bugged or not:

```
ag -uG '^(?!.*\.svn).*$' PATTERN
```

**Comment by cup** [comment]
https://github.com/ggreer/the_silver_searcher/issues/790

**Comment by nu8** [comment]
https://github.com/ggreer/the_silver_searcher/issues/245
https://github.com/ggreer/the_silver_searcher/issues/790
https://github.com/ggreer/the_silver_searcher/issues/917

**Comment by nu8** [comment]
Another workaround is:

~~~
ag --hidden --ignore .git
~~~

**Comment by ElSamhaa** [comment]
Any update on this?
